src/dags/bbga.py

```diff
@@ -0,0 +1,356 @@
+import csv
+import hashlib
+import itertools
+import shutil
+import sys
+import tempfile
+from csv import Dialect
+from dataclasses import dataclass
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import Callable, Dict, Tuple, TypedDict, Type, Any
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.postgres_operator import PostgresOperator
+from airflow.operators.python_operator import PythonOperator
+from common import DATAPUNT_ENVIRONMENT, MessageOperator, default_args, slack_webhook_token
+from http_fetch_operator import HttpFetchOperator
+from pgcomparator_cdc_operator import PgComparatorCDCOperator
+from postgres_create_tables_like_operator import PostgresCreateTablesLikeOperator
+from postgres_insert_csv_operator import FileTable, PostgresInsertCsvOperator
+from sqlalchemy_create_object_operator import SqlAlchemyCreateObjectOperator
+
+FileStem = str
+UrlPath = str
+
+DAG_ID = "bbga"
+TMP_DIR_BASE = Path("/tmp")
+TMP_TABLE_PREFIX = "tmp_"
+VARS = Variable.get(DAG_ID, deserialize_json=True)
+data_endpoints: Dict[FileStem, UrlPath] = VARS["data_endpoints"]
+table_mappings: Dict[FileStem, str] = VARS["table_mappings"]
+
+assert set(data_endpoints.keys()) == set(
+    table_mappings.keys()
+), "Both mappings should have the same set of keys."
+
+
+class bbga_dialect(Dialect):
```

Comment:
BBGADialect als class name?

Reply:
I have followed the convention of the existing Dialect names in module `csv`:

```python
class excel(Dialect):
    ...

class excel_tab(excel):
    ...

class unix_dialect(Dialect):
    ...
```

Hence the all lower case dialect name `bbga_dialect`.

But it does make me wonder what the naming convention is for our projects. I've grown used to only capitalizing the first letter of each abbreviation. So had I used the a more Python like naming style, I would have used `BbgaDialect` instead of `BBGADialect` to prevent too many capital letters. This is nicely demonstrated in names such as `XmlHttpRequest` that become much more unreadable when written as XMLHTTPRequest`.

This conventions is nicely motivated is this document: https://nim-doc.readthedocs.io/en/latest/nep1.html#naming-conventions

> In the age of HTTP, HTML, FTP, TCP, IP, UTF, WWW it is foolish to pretend these are somewhat special words requiring all uppercase. Instead treat them as what they are: Real words. So it’s parseUrl rather than parseURL, checkHttpHeader instead of checkHTTPHeader etc.